azurelinuxagent/ga/exthandlers.py

```diff
@@ -310,11 +309,68 @@ def handle_ext_handlers(self, etag=None):
                 logger.info("Extension handling is on hold")
                 return
 
+        dep_level = None
+        deps_wait_until = None
+        handlers_wait_until = datetime.datetime.utcnow()
+        prev_handler = None
+
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
+            # Use updated time limit for each dependency level
+            if dep_level != ext_handler.sort_key():
```

Comment:
Why not simply move through the extHandlers sorted by dependency level, and remove all the additional wait code?

Reply:
We not only make sure the extensions are installed sequentially, we need to make sure that the previous extension in the sequence is installed successfully before proceeding with the next extension in the sequence. Hence this wait code with different timeout value for each level.

Reply:
Consider something like this:

```
  for ext_handler in extHandlers:
    handle_ext_handler(ext_handler)
    wait_if_necessary(ext_handler)
```

Seems a little bit simpler than tracking full dependencies per se. We really just care if the last handler is done, if there is a stated dependency.

Regarding the timeouts, I thought we had decided on a total 90?

Reply:
Agree with @hglkrijger that the for loop can be simplified to not make reference to the dependencies, but just ext_handler. Are the dependencies used only to increase the timeout 90 min per dependency level?

Reply:
Even now, the for loop does not make any reference to the dependency. It just references what was the previous extension installed in order to wait on that.

I think the one @hglkrijger suggests is that 'install an extension, wait for it'.This has a side effect of waiting for the highest level extensions as well, which is not required to be waited for.

Another way based on his suggestion is 'install an extension, wait for it if it is not the highest dependency level'. This requires calculating the highest dependency level in advance.
Do you think we need to change the logic to match this?

Reply:
per offline discussion we will got for a flat 90 min